js/source/tile.js

```diff
@@ -129,11 +129,11 @@ class Tile {
     }
 
     redoPlacement(source) {
-        if (source.type !== 'vector' && source.type !== 'geojson') {
+        if ((source.type !== 'vector' && source.type !== 'geojson') ||
+            !this.hasData() || !this.collisionTile) {
```

Comment:
`redoWhenDone` is also used when the tile is loaded, [here](https://github.com/mapbox/mapbox-gl-js/blob/fb173257e734cffa04f6636bc03b3798bdcd8a4f/js/source/vector_tile_source.js#L91-L94).